internal/rbd/rbd_attach.go

```diff
@@ -250,6 +250,23 @@ func appendDeviceTypeAndOptions(cmdArgs []string, isNbd, isThick bool, userOptio
 	}
 
 	cmdArgs = append(cmdArgs, "--device-type", accessType)
+	if userOptions != "" {
+		// userOptions is appended after, possibly overriding the above
+		// default options.
+		cmdArgs = append(cmdArgs, "--options", userOptions)
+	}
+
+	isMap := true
+	for _, item := range cmdArgs {
+		if item == "unmap" {
+			isMap = false
+		}
+	}
+
+	if !isMap {
```

Comment:
A comment would be in place here, explaining that the options for `rbd-nbd` are only needed while mapping the device.

Just below there is a comment mentioning `unmapping images`, and passing the `noudev` option. Is that really not needed while unmapping?

Reply:
Yes looks like `noudev` is used with `unmap` too in some of ceph test case. Let's not touch it.

I have fixed this now. Thanks!